File: harmoni_sehat_backend/models/Dokter.js
const mongoose = require('mongoose');

const DokterSchema = new mongoose.Schema({
  user_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    unique: true,
  },
  nama: {
    type: String,
    required: true,
  },
  nomor_str: {
    type: String,
    unique: true,
  },
  spesialisasi_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Spesialisasi', // Assuming Spesialisasi model will be created
  },
  biaya_konsultasi: {
    type: Number,
  },
  foto: {
    type: String,
  },
  bio: {
    type: String,
  },
  status: {
    type: String,
    enum: ['active', 'inactive', 'pending'],
    default: 'pending',
  },
});

module.exports = mongoose.model('Dokter', DokterSchema);

File: harmoni_sehat_frontend/lib/shared/models/dokter.dart
class Dokter {
  final String id;
  final String nama;
  final String nomorStr;
  final double biayaKonsultasi;
  final String? spesialisasiId;
  final String? foto;
  final String? bio;

  Dokter({
    required this.id,
    required this.nama,
    required this.nomorStr,
    required this.biayaKonsultasi,
    this.spesialisasiId,
    this.foto,
    this.bio,
  });

  Map<String, dynamic> toJson() {
    return {
      'user_id': id,
      'nama': nama,
      'nomor_str': nomorStr,
      'biaya_konsultasi': biayaKonsultasi,
      if (spesialisasiId != null) 'spesialisasi_id': spesialisasiId,
      if (foto != null) 'foto': foto,
      if (bio != null) 'foto': foto,
    };
  }
}